### PR TITLE
Reset user default group when link between user and group is deleted

### PR DESCRIPTION
**Pull request by RomainLvr**
## Checklist before requesting a review

*Please delete options that are not relevant.*

- [x] I have read the CONTRIBUTING document.
- [x] I have performed a self-review of my code.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [ ] This change requires a documentation update.

## Description

- It fixes !35654
- Here is a brief description of what this PR does

When you delete the link between a user and a group, if this group is the user's default group, it remains so even after the link has been deleted.

## Screenshots (if appropriate):

396580549-7cc06cfc-e3eb-4d07-a33e-f93b6b268864

### CODE REVIEW

File: src/Group_User.php [hunk 1]
```diff
@@ -954,8 +954,20 @@ public function post_purgeItem()
...
+            $user_inst->update(
+                [
+                    'id'        => $users_id,
+                    'groups_id' => 0,
+                ]
+            );
+        }
+
        // find users with the current group in their plannings
         $user_inst = new User();
```
Comment (stonebuzz) [inline src/Group_User.php:972]: Instead of creating a new instance for the second operation, the same instance should be used, reducing redundancy.

**Review by stonebuzz — CHANGES_REQUESTED** [review]

File: phpunit/functional/Group_UserTest.php [hunk 2]
```diff
@@ -213,4 +213,48 @@ public function testIsUserInGroup()
...
+
+        $this->assertTrue(
+            $user->update(
+                [
+                    'id' => $user->getID(),
+                    'groups_id' => $gid
+                ]
+            )
+        );
+
```
Comment (trasher) [inline phpunit/functional/Group_UserTest.php:246]: Is that update really needed?
```suggestion
        $this->assertTrue(
            $user->update(
                [
                    'id' => $user->getID(),
                    'groups_id' => $gid
                ]
            )
        );

```

Comment (stonebuzz) [reply phpunit/functional/Group_UserTest.php:246]: yes, to set user default group =)

This is the link between the user and the group that was added previously, and this action does not update the default group

**Review by stonebuzz — APPROVED** [review]

**Review by cedric-anne — APPROVED** [review]
I just pushed a commit to move the `$planning_k` definition line at the same place it is used.

**Review by trasher — APPROVED** [review]

**Pull request merged**